src/controllers/post.js

```diff
@@ -1,39 +1,51 @@
 import dbClient from '../utils/dbClient.js'
-
 import { sendDataResponse } from '../utils/responses.js'
+import Joi from 'joi'
 
 export const create = async (req, res) => {
-  const { content } = req.body
-  if (!content) {
-    return sendDataResponse(res, 400, { content: 'Must provide content' })
+  const schema = Joi.object({
+    content: Joi.string().min(3).required()
+  })
+  const { error, value } = schema.validate(req.body)
+  if (error) {
+    return sendDataResponse(res, 400, { error: error.details[0].message })
   }
-  const createdPost = await dbClient.post.create({
-    data: {
-      content: content,
-      user: {
-        connect: {
-          id: req.user.id
+  const { content } = value
+  try {
+    const createdPost = await dbClient.post.create({
+      data: {
+        content: content,
+        user: {
+          connect: {
+            id: req.user.id
+          }
         }
       }
-    }
-  })
-  return sendDataResponse(res, 201, { post: createdPost })
+    })
+    return sendDataResponse(res, 201, { post: createdPost })
+  } catch (e) {
+    return sendDataResponse(res, 500, { error: e.message })
+  }
 }
 
 export const createComment = async (req, res) => {
-  const { content } = req.body
-  const { postId } = req.params
-
-  if (!content) {
-    return sendDataResponse(res, 400, { content: 'Must provide content' })
+  const schema = Joi.object({
+    content: Joi.string().min(3).max(150).required()
+  })
+  const { error, value } = schema.validate(req.body)
+  if (error) {
+    return sendDataResponse(res, 400, { error: error.details[0].message })
   }
-
-  if (!postId) {
+  const { content } = value
+  const { postId } = req.params
+  const commentOnPost = await dbClient.post.findUnique({
+    where: { id: parseInt(postId) }
+  })
+  if (!commentOnPost) {
     return sendDataResponse(res, 400, {
-      post: 'comments must be related to a post'
+      post: 'You CANNOT comment on a nonexistent post'
```

Comment:
It might be worth sticking to a common format for error responses? I.e:

```javascript
return sendDataResponse(res, 400, { error: 'Post not found'} )
```

Minor, but 404 may be a more suitable error?

Reply:
yeah sure it is more logical too, will do it